### PR TITLE
REF: move methods from base to Index

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by topper-123** [comment]
I've merged #52428, so I think this should be rebased to avoid merge issues here.

File: pandas/core/series.py [hunk 1]
```diff
@@ -5395,10 +5396,8 @@ def _convert_dtypes(
         return result
 
     # error: Cannot determine type of 'isna'
-    # error: Return type "Series" of "isna" incompatible with return type "ndarray
-    # [Any, dtype[bool_]]" in supertype "IndexOpsMixin"
     @doc(NDFrame.isna, klass=_shared_doc_kwargs["klass"])  # type: ignore[has-type]
-    def isna(self) -> Series:  # type: ignore[override]
+    def isna(self) -> Series:
         return NDFrame.isna(self)
```
Comment (topper-123) [inline pandas/core/series.py:5401]: `return super().isna()`?

Same question below.

**Review by topper-123 — COMMENTED** [review]
Can `return ndframe.method(self, ...)` be replaced with `return super().method(...)` for many methods here?

File: pandas/core/series.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/series.py:5401]: avoiding super bc i always have to check which parent class it refers to

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**